electron-app/src/websites/artconomy/artconomy.service.ts

```diff
@@ -0,0 +1,239 @@
+import { Injectable } from '@nestjs/common';
+import UserAccountEntity from 'src/account/models/user-account.entity';
+import { UsernameParser } from 'src/description-parsing/miscellaneous/username.parser';
+import ImageManipulator from 'src/file-manipulation/manipulators/image.manipulator';
+import Http, {HttpResponse} from 'src/http/http.util';
+import { SubmissionRating } from 'src/submission/enums/submission-rating.enum';
+import { FileSubmissionType } from 'src/submission/file-submission/enums/file-submission-type.enum';
+import { FileRecord } from 'src/submission/file-submission/interfaces/file-record.interface';
+import { FileSubmission } from 'src/submission/file-submission/interfaces/file-submission.interface';
+import { CancellationToken } from 'src/submission/post/cancellation/cancellation-token';
+import { FilePostData } from 'src/submission/post/interfaces/file-post-data.interface';
+import { PostResponse } from 'src/submission/post/interfaces/post-response.interface';
+import { DefaultOptions } from 'src/submission/submission-part/interfaces/default-options.interface';
+import { SubmissionPart } from 'src/submission/submission-part/interfaces/submission-part.interface';
+import { ValidationParts } from 'src/submission/validator/interfaces/validation-parts.interface';
+import FileSize from 'src/utils/filesize.util';
+import WebsiteValidator from 'src/utils/website-validator.util';
+import { LoginResponse } from '../interfaces/login-response.interface';
+import { ScalingOptions } from '../interfaces/scaling-options.interface';
+import { UsernameShortcut } from '../interfaces/username-shortcut.interface';
+import { Website } from '../website.base';
+import { ArtconomyAccountData } from './artconomy-account.interface';
+import { ArtconomyDefaultFileOptions } from './artconomy.defaults';
+import { ArtconomyOptions } from './artconomy.interface';
+import {MarkdownParser} from "src/description-parsing/markdown/markdown.parser";
+
+
+@Injectable()
+export class Artconomy extends Website {
+  readonly BASE_URL: string = 'https://artconomy.com';
+  readonly acceptsFiles: string[] = [
+    'png',
+    'jpeg',
+    'jpg',
+    'gif',
+    'mp4',
+    'doc',
+    'rtf',
+    'txt',
+    'mp3',
+  ];
+  readonly defaultDescriptionParser = MarkdownParser.parse;
+  readonly fileSubmissionOptions: object = ArtconomyDefaultFileOptions;
+  usernameShortcuts: UsernameShortcut[] = [
+    {
+      key: 'ac',
+      url: 'https://artconomy/$1',
+    },
+  ];
+
+  async checkLoginStatus(data: UserAccountEntity): Promise<LoginResponse> {
+    const status: LoginResponse = { loggedIn: false, username: null};
+    if (!data.data && data.data.authtoken) {
```

Comment:
Nitpick: Prefer to not have an early termination here.

Just wrap the below code in ```if (data.data && data.data.authtoken)```